rtmpsuck.c

```diff
@@ -776,9 +774,23 @@ TFTYPE doServe(void *arg)	// server socket and state (our listening socket)
     {
       if (!RTMPPacket_IsReady(&ps))
         continue;
-      ServePacket(server, 0, &ps);
+      /* Client can set chunk_size before the connect command */
+      if (ps.m_packetType == RTMP_PACKET_TYPE_CHUNK_SIZE)
+        {
+          if (ps.m_nBodySize >= 4)
+            {
+              server->rs.m_inChunkSize = AMF_DecodeInt32(ps.m_body);
+              RTMP_Log(RTMP_LOGDEBUG, "%s, client: chunk size change to %d", __FUNCTION__,
+                       server->rs.m_inChunkSize);
+            }
+        }
+      else
+        {
+          ServePacket(server, 0, &ps);
+          connect_cmd_processed = TRUE;
```

Comment:
It is tracking the state of the connect command in context of the RTMP protocol.  As I wrote, ServePacket does only call ServeInvoke, and the rest of invoke commands handled by the func. must be preceded by the connect command.  The variable can be either added into the STREAMING_SERVER structure or propagated as parameter into ServeInvoke.  But I don't think it is necessary, if considering the context.  A solid one line comment might help as well,  but I'm not sure it is necessary.